File: 1_50/12_integer_to_roman.c
char* intToRoman(int num) {
    int number[] = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
    char roman[][3] = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX",
                      "V", "IV", "I"};
    int n = 13;
    char *arr = (char*)malloc(sizeof(char) * 50);
    
    while (num != 0) {
        for (int i=0; i<n; i++) {
            while (num >= number[i]) {
                strcat(arr, roman[i]);
                num -= number[i];
            }
        }
    }
    return arr;
}

File: 1_50/12_integer_to_roman.py
class Solution:
    def intToRoman(self, num):
        """
        :type num: int
        :rtype: str
        """
        nums = [(1000, 'M'), (900, 'CM'), (500, 'D'), (400, 'CD'), (100, 'C'),
               (90, 'XC'), (50, 'L'), (40, 'XL'), (10, 'X'), (9, 'IX'),
               (5, 'V'), (4, 'IV'), (1, 'I')]
        roman = ""
        
        # % operator is slow
        while num != 0:
            for (n, c) in nums:
                while num >= n:
                    roman += c
                    num -= n

        return roman
